recipes/qwt/all/conanfile.py

```diff
@@ -0,0 +1,113 @@
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _patch_qwt_config_files(self):
+        # qwtconfig.pri
+        qwtconfig_path = os.path.join(self.source_folder, self._source_subfolder, "qwtconfig.pri")
+        qwtconfig = tools.load(qwtconfig_path)
+
+        qwtconfig = "CONFIG += conan_basic_setup\ninclude(../conanbuildinfo.pri)\n" + qwtconfig
+        qwtconfig += "QWT_CONFIG {}= QwtDll\n".format("+" if self.options.shared else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtPlot\n".format("+" if self.options.plot else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtWidgets\n".format("+" if self.options.widgets else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtSvg\n".format("+" if self.options.svg else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtOpenGL\n".format("+" if self.options.opengl else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtMathML\n".format("+" if self.options.mathml else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtDesigner\n".format("+" if self.options.designer else "-")
+        tools.save(qwtconfig_path, qwtconfig)
+
+        # qwtbuild.pri
+        qwtbuild_path = os.path.join(self.source_folder, self._source_subfolder, "qwtbuild.pri")
+        qwtbuild = tools.load(qwtbuild_path)
+        # set build type
+        qwtbuild += "CONFIG -= debug_and_release\n"
+        qwtbuild += "CONFIG -= build_all\n"
+        qwtbuild += "CONFIG -= release\n"
+        qwtbuild += "CONFIG += {}\n".format("debug" if self.settings.build_type == "Debug" else "release")
+        if self.settings.build_type == "RelWithDebInfo":
+            qwtbuild += "CONFIG += force_debug_info\n"
+        tools.save(qwtbuild_path, qwtbuild)
+
+    def build(self):
+        self._patch_qwt_config_files()
+
+        if self.settings.compiler == "Visual Studio":
+            vcvars = tools.vcvars_command(self.settings)
+            self.run("{} && qmake {}".format(vcvars, self._source_subfolder), run_environment=True)
+            self.run("{} && nmake".format(vcvars))
+        else:
+            self.run("qmake {}".format(self._source_subfolder), run_environment=True)
+            self.run("make -j {}".format(tools.cpu_count()))
+
+    def package(self):
+        self.copy("COPYING", src=os.path.join(self._source_subfolder), dst="licenses")
+        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "src"))
+        self.copy("*.dll", dst="bin", keep_path=False)
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.so*", dst="lib", keep_path=False, symlinks=True)
+        self.copy("*.dylib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.includedirs = ['include']
+        if self.settings.build_type == "Debug":
+            self.cpp_info.libs = ["qwtd"] if self.settings.os == "Windows" else ["qwt_debug"]
```

Comment:
@boussaffawalid this should improve the situation (not tested)
```suggestion
            if self.settings.os == "Windows":
                self.cpp_info.libs = ["qwtd"]
            elif self.settings.os == "Macos":
                self.cpp_info.libs = ["qwt_debug"]
            else:
                self.cpp_info.libs = ["qwt"]
```